collectd/conf.go

```diff
@@ -229,22 +238,22 @@ func validatedCollectdConfig(metrics *Metrics) (*collectdConf, error) {
 	}
 	for pipelineID, pipeline := range metrics.Service.Pipelines {
 		if strings.HasPrefix(pipelineID, "lib:") {
-			return nil, fmt.Errorf(`pipeline id prefix 'lib:' is reserved for pre-defined pipelines. Pipeline ID %q is not allowed.`, pipelineID)
+			return nil, reservedIdPrefixError("pipeline", pipelineID)
 		}
 		if len(pipeline.ReceiverIDs) != 1 {
 			return nil, errors.New("exactly one receiver id is required in the metrics service pipeline receiver id list.")
 		}
 		invalidReceiverIDs := findInvalid(definedReceiverIDs, pipeline.ReceiverIDs)
 		if len(invalidReceiverIDs) > 0 {
-			return nil, fmt.Errorf("metrics receivers not defined: %v", invalidReceiverIDs)
+			return nil, fmt.Errorf("metrics receiver %q from pipeline %q is not defined.", invalidReceiverIDs[0], pipelineID)
 		}
 
 		if len(pipeline.ExporterIDs) != 1 {
 			return nil, errors.New("exactly one exporter id is required in the metrics service pipeline exporter id list.")
 		}
 		invalidExporterIDs := findInvalid(definedExporterIDs, pipeline.ExporterIDs)
 		if len(invalidExporterIDs) > 0 {
-			return nil, fmt.Errorf("metrics exporters not defined: %v", invalidExporterIDs)
+			return nil, fmt.Errorf("metrics exporter %q from pipeline %q is not defined.", invalidExporterIDs, pipelineID)
```

Comment:
Here, the messages says "exporter" and not "exporters" but we are printing the whole list of them.  I think this should look similar to the one above, on line 248, and they should do something consistent. WDYT?

Reply:
Oops. Nice catch! Yeah, keeping them consistent was the intention. Just fixed.